main/src/http/http_service.rs

```diff
@@ -1257,6 +1358,112 @@ fn construct_write_tsdb_points_json_request(req: &Bytes) -> Result<Vec<Line>, Ht
+}
+
+fn try_parse_es_req(req: &Bytes) -> Result<Vec<ESLog>, HttpError> {
+    let mut eslogs = vec![];
+
+    let lines = simdutf8::basic::from_utf8(req.as_ref())
+        .map_err(|e| HttpError::InvalidUTF8 { source: e })?;
+
+    let json_chunk: Vec<&str> = lines.trim().split('\n').collect();
+    let mut i = 0;
+    let n = json_chunk.len();
+
+    if n % 2 != 0 {
+        println!("error syntax: {:?}", req);
+        return Err(HttpError::ParseESLog {
+            source: protocol_parser::ESLogError::InvaildSyntax,
+        });
+    }
+
+    while i < n {
+        let command: Command = serde_json::from_str(json_chunk[i])
+            .map_err(|e| HttpError::ParseESLogJson { source: e })?;
+        let fields: Fields = serde_json::from_str(json_chunk[i + 1])
+            .map_err(|e| HttpError::ParseESLogJson { source: e })?;
+
+        eslogs.push(ESLog { fields, command });
+
+        i += 2;
+    }
+
+    Ok(eslogs)
+}
+
+async fn coord_write_eslog(
+    coord: &CoordinatorRef,
+    tenant: &str,
+    db: &str,
+    table: &str,
+    es_log: Vec<ESLog>,
+    time_alias: Option<String>,
+    msg_alias: Option<String>,
+    span_context: Option<&SpanContext>,
+) -> Result<warp::reply::Json, HttpError> {
+    let mut span_recorder = SpanRecorder::new(span_context.child_span("write points"));
+    let mut res: HashMap<String, String> = HashMap::new();
+
+    let table_exist = {
+        if let Some(meta) = coord.meta_manager().tenant_meta(tenant).await {
+            meta.get_table_schema(db, table).is_ok()
+        } else {
+            return Err(HttpError::ParseESLog {
+                source: protocol_parser::ESLogError::InvaildSyntax,
+            });
+        }
+    };
+    let mut lines = Vec::new();
+
+    let time_key = time_alias.unwrap_or_default();
+    let msg_key = msg_alias.unwrap_or_default();
+
+    for (i, es) in es_log.iter().enumerate() {
+        let line = es_parse_to_line(es, table, &time_key, &msg_key)
+            .map_err(|e| HttpError::ParseESLog { source: e })?;
+        match es.command {
+            Command::Create(_) => {
+                if table_exist {
+                    res.insert(format!("{}-create", i), "table has exist".to_string());
+                    break;
+                }
+                lines.push(line);
+                res.insert(format!("{}-create", i), "Success!".to_string());
+            }
+            Command::Index(_) => {
+                lines.push(line);
+                res.insert(format!("{}-index", i), "Success!".to_string());
+            }
+            Command::Invalid => {
+                res.insert(format!("{}-command", i), "invalid command".to_string());
+                break;
+            }
+        }
+    }
+
+    coord
+        .write_lines(tenant, db, Precision::MS, lines, span_recorder.span_ctx())
+        .await
+        .map_err(|e| {
+            span_recorder.error(e.to_string());
+            e
+        })?;
+
+    Ok(warp::reply::json(&res))
```

Comment:
Here, the error message tells us which line was wrong, and we need to return it if the write was successful:
```
HTTP/1.1 200 OK
content-length: 0
date: Sat, 08 Oct 2022 06:59:38 GMT
```